Apps/W1/EDocumentConnectors/Continia/app/src/API Requests/APIRequests.Codeunit.al

```diff
@@ -0,0 +1,1107 @@
+        HttpRequest.SetRequestUri(Url);
+
+        if HttpContentData <> '' then begin
+            HttpContent.WriteFrom(HttpContentData);
+            HttpContent.GetHeaders(HttpContentHeaders);
+            if HttpContentHeaders.Contains('Content-Type') then
+                HttpContentHeaders.Remove('Content-Type');
+            HttpContentHeaders.Add('Content-Type', 'application/xml');
+            HttpRequest.Content := HttpContent;
+        end;
+
+        if HttpClient.Send(HttpRequest, HttpResponse) then
+            exit(HandleApiError(HttpResponse));
+    end;
+
+    internal procedure HandleApiError(var HttpResponse: HttpResponseMessage): Boolean
+    var
+        HttpStatusCode: Integer;
+        ResponseBody: Text;
+        ResponseBodyXML: XmlDocument;
+    begin
+        HttpStatusCode := HttpResponse.HttpStatusCode;
+        if HttpStatusCode in [200, 201, 202] then
+            exit(true);
+
+        HttpResponse.Content().ReadAs(ResponseBody);
+        if ResponseBody <> '' then
+            if not XmlDocument.ReadFrom(ResponseBody, ResponseBodyXML) then
+                exit(ThrowError(UnexpectedApiErr));
+
+        ReadErrorResponse(ResponseBodyXML, HttpStatusCode);
+    end;
+
+    internal procedure ReadErrorResponse(var ResponseXML: XmlDocument; HttpStatusCode: Integer): Boolean
+    var
+        ApiErrorCode: Text;
+        ApiErrorMessage: Text;
+        ErrorCodeNode: XmlNode;
+        ErrorMessageNode: XmlNode;
+    begin
+        if not ResponseXML.SelectSingleNode('/error/code', ErrorCodeNode) then
+            exit(ThrowError(UnexpectedApiErr));
+
+        if not ResponseXML.SelectSingleNode('/error/message', ErrorMessageNode) then
+            exit(ThrowError(UnexpectedApiErr));
+
+        ApiErrorCode := ErrorCodeNode.AsXmlElement().InnerText;
+        ApiErrorMessage := ErrorMessageNode.AsXmlElement().InnerText;
+
+        case HttpStatusCode of
+            500, 501:
+                exit(ThrowError(StrSubstNo(Error500Err, ApiErrorCode, ApiErrorMessage)));
+            400, 401, 402, 404, 409, 422:
+                exit(ThrowError(StrSubstNo(ApiErr, ApiErrorCode, ApiErrorMessage)));
+            else
+                exit(ThrowError(UnexpectedApiErr));
+        end;
+    end;
+
+    local procedure ThrowError(ErrorMsg: Text): Boolean
+    begin
+        if SuppressError then
+            exit;
+
+        Error(ErrorMsg);
+    end;
+
+    local procedure GetCurrentCountryCode(): Code[10]
+    var
+        CompanyInformation: Record "Company Information";
+        CountryRegion: Record "Country/Region";
+    begin
+        CompanyInformation.Get();
+        if CompanyInformation."Country/Region Code" = '' then
+            exit('');
+
+        if CountryRegion.Get(CompanyInformation."Country/Region Code") then
+            exit(CountryRegion."ISO Code");
+    end;
+
+    internal procedure GetGuidAsText(Value: Guid): Text[36]
+    begin
+        exit(CopyStr(DelChr(Value, '<>', '{}'), 1, 36))
```

Comment:
nit: This might be easier with the Format functionality:
Format(Value, 0, 4) returns a guid without curly braces.
https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/devenv-format-property#standard-guid-formats